src/libmongoc/src/mongoc/mongoc-database.c

```diff
@@ -1100,15 +1098,37 @@ create_collection_with_encryptedFields (mongoc_database_t *database,
    bool state_collections_ok =
       create_encField_state_collection (
          database, encryptedFields, name, "esc", error) &&
-      create_encField_state_collection (
-         database, encryptedFields, name, "ecc", error) &&
       create_encField_state_collection (
          database, encryptedFields, name, "ecoc", error);
    if (!state_collections_ok) {
       // Failed to create one or more state collections
       goto fail;
    }
 
+   // Check the wire version to ensure server is 7.0.0 or newer.
+   {
+      mongoc_server_stream_t *stream =
+         mongoc_cluster_stream_for_writes (&database->client->cluster,
+                                           NULL /* client session */,
+                                           NULL /* reply */,
+                                           error);
+      if (!stream) {
+         goto fail;
+      }
+      if (stream->sd->max_wire_version < WIRE_VERSION_7_0) {
+         bson_set_error (
+            error,
+            MONGOC_ERROR_PROTOCOL,
+            MONGOC_ERROR_PROTOCOL_BAD_WIRE_VERSION,
+            "Driver support of Queryable Encryption is incompatible "
+            "with server. Upgrade server to use Queryable Encryption.");
```

Comment:
Suggest including both the detected max wire version and the required server version in the error message.